mirest/ah-haven-space-sprinters
# #164694239 Users can report an article

#### What does this PR do?
* Add report an article functionality

#### Description of Task to be completed?

- [x] add ReportArticle model and related serializer classes

- [x] add ReportArticle to the admin site

- [x] add ReportArticle views and relevant methods

- [x] write unittests

#### How should this be manually tested?
You can test locally by running command `python manage.py runserver`
- Log in using verified credentials
- Go to article of your choice, you can report an article to using this endpoint `POST/api/articles/:slug/report` and json object `{"body":"value"}`
- To view the reports an admin should either use the endpoint  `POST/api/articles/:slug/report` or login using the admin site and navigate to ReportArticle

#### Relevant screenshots of task
authenticated user reporting an article
![image](https://user-images.githubusercontent.com/44563961/56726731-062e6e80-6758-11e9-9ad8-1bdb60b7bc47.png)


retriving all reports by admin user
![image](https://user-images.githubusercontent.com/44563961/56726353-3aedf600-6757-11e9-82ff-891adb8b90e1.png)


#### What are the relevant pivotal tracker stories?
[#164694239](https://www.pivotaltracker.com/story/show/164694239)
